src/ucp/rma/flush.c

```diff
@@ -449,19 +469,80 @@ static void ucp_worker_flush_ep_flushed_cb(ucp_request_t *req)
     ucp_request_put(req);
 }
 
+static UCS_F_ALWAYS_INLINE ucp_ep_h
+ucp_worker_flush_req_get_ep(ucp_request_t *req)
+{
+    ucp_worker_h worker           = req->flush_worker.worker;
+    ucs_list_link_t *next_ep_iter = NULL;
+    ucp_ep_h ep;
+
+    ucs_assert(req->flush_worker.ep_iter != NULL);
+
+    do {
+        if (req->flush_worker.ep_iter != &worker->all_eps) {
+            if (next_ep_iter == NULL) {
+                next_ep_iter = req->flush_worker.ep_iter->next;
```

Comment:
this became even more complicated now

Reply:
is it ok to move to the previous implementation?

Reply:
i'd expect it to be even simpler than prev implementation